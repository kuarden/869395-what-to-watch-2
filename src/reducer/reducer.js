import {films} from '../mocks/films.js';

export const getFilms = (genre) => {
  if (genre === `All genres`) {
    return films.slice();
  }
  return films.slice().filter((n) => n.genre === genre);
};

export const initialState = {
  genre: `All genres`,
  films,
};

export const ActionCreator = {
  changeGenre: (genre) => ({
    type: `CHANGE_GENRE`,
    payload: genre,
  }),

  getFilms: (genre) => ({
    type: `GET_FILMS`,
    payload: getFilms(genre),
  })
};

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `CHANGE_GENRE`:
      return Object.assign({}, state, {
        genre: action.payload,
      });
    case `GET_FILMS`:
      return Object.assign({}, state, {
        films: action.payload,
      });
  }

  return state;
};
