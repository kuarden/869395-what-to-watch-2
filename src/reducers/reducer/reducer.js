import {combineReducers} from "redux";
import {data} from "../data/data";
import {user} from "../user/user";

export const reducer = combineReducers({data, user});
