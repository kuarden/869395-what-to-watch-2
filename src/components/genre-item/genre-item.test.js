import React from "react";
import renderer from "react-test-renderer";
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import {reducer} from "../../reducer/reducer.js";
import GenreItem from "./genre-item.jsx";

it(`GenreItem renders correctly`, () => {
  const store = createStore(reducer);
  const tree = renderer
    .create(<Provider store={store}><GenreItem
      genreName={``}
      activeTab={false}
      onGenreClick={() => {}}
    /></Provider>).toJSON();

  expect(tree).toMatchSnapshot();
});
