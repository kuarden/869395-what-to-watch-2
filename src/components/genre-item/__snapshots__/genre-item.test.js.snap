// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreItem renders correctly 1`] = `
<li
  className="catalog__genres-item"
>
  <a
    className="catalog__genres-link"
    href="#"
    onClick={[Function]}
  />
</li>
`;
